File: Python/Flamaster.py
def compress_string(inscription):
    compressed = inscription[0]
    count = 1
    for i in range(1, len(inscription)):
        if inscription[i] == inscription[i - 1]:
            count += 1
        else:
            if count > 2:
                compressed += str(count)
            elif count == 2:
                compressed += inscription[i - 1]
            compressed += inscription[i]
            count = 1
    if count > 2:
        compressed += str(count)
    elif count == 2:
        compressed += inscription[-1]
    return compressed

C = int(input())
for _ in range(C):
    inscription = input()
    print(compress_string(inscription))

File: Java/Flamaster.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int C = input.nextInt();
        input.nextLine();
        for (int i = 0; i < C; i++) 
        {
            String string = input.nextLine();
            compressString(string);
        }
        input.close();
    }

    private static void compressString(String string) 
    {
        StringBuilder compressed = new StringBuilder();
        char repeatedLetter = string.charAt(0);
        int count = 1;
        for (int i = 1; i < string.length(); i++) 
        {
            char currentLetter = string.charAt(i);
            if (currentLetter == repeatedLetter) 
                count++;
            else 
            {
                compressed.append(repeatedLetter);
                if (count > 2) 
                    compressed.append(count);
                else if (count == 2) 
                    compressed.append(repeatedLetter);
                repeatedLetter = currentLetter;
                count = 1;
            }
        }
        compressed.append(repeatedLetter);
        if (count > 2) 
            compressed.append(count);
        else if (count == 2) 
            compressed.append(repeatedLetter);
        System.out.println(compressed);
    }
}